### DISCUSSION TITLE
"Right" location for an external DLL

### DISCUSSION THREAD

**Discussion by dn89000** [gaelcolas/Sampler#429] [Q&A] [answered]
I'm writing a custom PowerShell module and I need to distribute a DLL with the code.  What is the "right" directory to put the DLL in using the SimpleModule scaffold?

**Comment by mikebijl** [comment]
This is a very good question... I'm also looking into a (preferably) hybrid module with C# based cmdlets and Powershell based functions. Is it possible to include that in the framework, too?

**Comment by johlju** [comment, marked as answer]
This was answered in the discussion https://github.com/gaelcolas/Sampler/discussions/459 (that was an issue that I just converted to a discussion without noticing there already was a discussion of the same topic):

**Comment by johlju** [comment]
@mikebijl could you please create a separate discussion topic for what you are looking for? I think it is an interesting topic.

**Reply by mikebijl** [reply]
Hi @johlju,

So not sure what you mean by creating a seperate discussion topic... this is a seperate discussion, right? But maybe that is because I'm not that familiar yet with Github and contributing. 

But now we are building our own module which is progressing quite ok. However there are some discussions within the company about "static code scanning" being a hard requirement on all generated code. So powershell and static code scanning is not the most advanced combo.

So I was wondering if I could incorporate this way of work [Creating your first PowerShell module in C# - Emanuel Palm,Justin Grote - PSConfEU 2023](https://youtu.be/6jQqf-LTRGI?si=dQTqHiVuRT1701D9) with the "functions" approach. This would give us a perfect sweet spot on moving function by function to the C# WoW.

Sorry for not making it a seperate discussion 😄
